NDB.Covid19/NDB.Covid19/WebServices/AnonymousTokenService.cs

```diff
@@ -0,0 +1,108 @@
+﻿using AnonymousTokens.Client.Protocol;
+using AnonymousTokens.Core.Services;
+using AnonymousTokens.Core.Services.InMemory;
+using NDB.Covid19.Configuration;
+using NDB.Covid19.OAuth2;
+using Newtonsoft.Json;
+using Org.BouncyCastle.Asn1.X9;
+using Org.BouncyCastle.Crypto.EC;
+using Org.BouncyCastle.Math.EC;
+using Org.BouncyCastle.Utilities.Encoders;
+using System;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NDB.Covid19.WebServices
+{
+    public class AnonymousTokenService
+    {
+        private Initiator _initiator = new Initiator();
+        private X9ECParameters _ecParameters = CustomNamedCurves.GetByOid(X9ObjectIdentifiers.Prime256v1);
```

Comment:
We discussed that this should be a part of a configuration. That would probably suffice for this case.

Potential later improvement: The curve info is included with the key. It is already implicitly so, but it is seemingly not straightforward to detect which curve it actually is, so it could introduce problems by using it directly, so it's good to have something to verify against. (Potential workarond: Keep a list of all accepted named curves, create objects for each in turn, and check that _ecParameters.Curve.Equals(givenCurve) for exactly one of them.